scripts/synthetix/facilitate_trade.js

```diff
@@ -51,6 +51,26 @@ const gasStationURL = {
   4: "https://safe-relay.rinkeby.gnosis.io/api/v1/gas-station/",
 }
 
+const estimationURLPrexix = {
+  1: "https://dex-price-estimator.gnosis.io//api/v1/",
+  4: "https://dex-price-estimator.rinkeby.gnosis.io//api/v1/",
+}
+
+const estimatePrice = async function (buyTokenId, sellTokenId, sellAmount, networkId) {
+  const searchCriteria = `markets/${buyTokenId}-${sellTokenId}/estimated-buy-amount/${sellAmount}?atoms=true`
+  const estimationData = await (await fetch(estimationURLPrexix[networkId] + searchCriteria)).json()
```

Comment:
nit: shouldn't need to await on json() as it's likekly synchronous.

Reply:
I needs to be there. I have dealt with this very same comment in the past.

Reply:
You are right, it's needed https://github.com/node-fetch/node-fetch/blob/fa627f4b0cc60570037c2fc8db88b1771ae185b4/example.js#L14

I assume the first promise returns as soon as there is the first bit of response data but json() requires the full response to be there so it is a separate promise. Clever, but a bit non-standard.

Reply:
Good stuff! I only hope I can remember how to find this link again later when it comes up again.